### PR TITLE
test: ensure that build URLs are valid

### PR DESCRIPTION
**Pull request by wchargin**
Summary:
This tests that URLs point to the TensorFlow mirror (as opposed to the
Bazel mirror) and that they resolve to valid archives.

Test Plan:
Run `./tensorboard/tools/mirror_urls_test.sh`, and note that it passes
with no output. Then apply the following patch to add some bad URLs:

```diff
diff --git a/WORKSPACE b/WORKSPACE
index 3943b0cf..d3975dfe 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -9,6 +9,9 @@ http_archive(
     urls = [
         # tag 0.7.0 resolves to commit 6741f733227dc68137512161a5ce6fcf283e3f58 (2019-02-08 18:37:26 +0100)
         "http://mirror.tensorflow.org/github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
+        "http://mirror.tensorflow.org/example.com/nonexistent.txt",
+        "http://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
+        "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
         "https://github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
     ],
 )
```

Re-run the test, and note that it fails with exit code 1 and this text:

```
The following URLs are not properly mirrored:
  - http://mirror.tensorflow.org/example.com/nonexistent.txt
Please comment on your PR asking a TensorBoard core team member
to mirror these URLs per instructions in http://b/133880558.

The following URLs point to the legacy Bazel mirror:
WORKSPACE:13:        "http://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
WORKSPACE:14:        "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/archive/0.7.0.tar.gz",
Please update them to use http://mirror.tensorflow.org/ instead.

```

Tested on Git v2.15.1, which is what Travis uses.

wchargin-branch: mirror-urls-test

### CODE REVIEW

File: tensorboard/tools/mirror_urls_test.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    rm "${bazel_urls_file}" || true
+}
+trap cleanup EXIT
+
+check_urls_resolve() {
+    unresolved_urls_file="$(mktemp)"
+    # shellcheck disable=SC2016
+    check_cmd='curl -sfL "$1" >/dev/null || printf "%s\n" "$1"'
+    # shellcheck disable=SC2016
+    exclude='${version}'  # as in `ci/download_bazel.sh`
```
Comment (nfelt) [inline tensorboard/tools/mirror_urls_test.sh]: Can we just exclude the specific ci/download_bazel.sh file itself?  That seems less brittle.

File: tensorboard/tools/mirror_urls_test.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    # list of URLs may potentially be mangled.
+
+    if ! [ -s "${unresolved_urls_file}" ]; then
+        return 0
+    fi
+
+    printf '%s\n' 'The following URLs are not properly mirrored:'
+    cat "${unresolved_urls_file}"
+    printf '%s\n' \
+        'Googlers, see http://b/133880558 for further instructions.' \
```
Comment (nfelt) [inline tensorboard/tools/mirror_urls_test.sh]: More important that instructions for Googlers I think are instructions for what to do as a non-core-TB-team member who gets this error when sending a PR.  Probably something like "please comment on your PR asking a TensorBoard core team member to mirror the URLs as per instructions in http://b/133880558".

File: tensorboard/tools/mirror_urls_test.sh (hunk 1 above)
Comment (wchargin) [reply tensorboard/tools/mirror_urls_test.sh]: Sure; done.

File: tensorboard/tools/mirror_urls_test.sh (hunk 2 above)
Comment (wchargin) [reply tensorboard/tools/mirror_urls_test.sh]: Sure; done.

**Pull request merged**